ProcessLib/SourceTerms/LineSourceTermFEM.h

```diff
@@ -0,0 +1,127 @@
+        double const t, GlobalVector& b) = 0;
+    virtual ~LineSourceTermLocalAssemblerInterface() = default;
+};
+
+const unsigned NUM_NODAL_DOF = 1;
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned GlobalDim>
+class LineSourceTermLocalAssembler final
+    : public LineSourceTermLocalAssemblerInterface
+{
+    using ShapeMatricesType = ShapeMatrixPolicyType<ShapeFunction, GlobalDim>;
+
+    using LocalAssemblerTraits = ProcessLib::LocalAssemblerTraits<
+        ShapeMatricesType, ShapeFunction::NPOINTS, NUM_NODAL_DOF, GlobalDim>;
+
+    using NodalVectorType = typename LocalAssemblerTraits::LocalVector;
+    using NodalRowVectorType = typename ShapeMatricesType::NodalRowVectorType;
+
+public:
+    LineSourceTermLocalAssembler(
+        MeshLib::Element const& element,
+        std::size_t const local_matrix_size,
+        bool is_axially_symmetric,
+        unsigned const integration_order,
+        ParameterLib::Parameter<double> const& line_source_term_parameter)
+        : _parameter(line_source_term_parameter),
+          _integration_method(integration_order),
+          _local_rhs(local_matrix_size)
+    {
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        auto const shape_matrices =
+                       initShapeMatrices<ShapeFunction, ShapeMatricesType,
+                                         IntegrationMethod, GlobalDim>(
+                           element, is_axially_symmetric, _integration_method);
+
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            _ip_data.emplace_back(
+                shape_matrices[ip].N,
+                _integration_method.getWeightedPoint(ip).getWeight() *
+                    shape_matrices[ip].integralMeasure *
+                    shape_matrices[ip].detJ);
+        }
+    }
+
+    void integrate(std::size_t const id,
+                   NumLib::LocalToGlobalIndexMap const& source_term_dof_table,
+                   double const t, GlobalVector& b) override
+    {
+        _local_rhs.setZero();
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        ParameterLib::SpatialPosition pos;
+        pos.setElementID(id);
+
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            pos.setIntegrationPoint(ip);
+            auto const st_val = _parameter(t, pos)[0];
+
+            _local_rhs.noalias() +=
+                st_val * _ip_data[ip].integration_weight_times_N;
+        }
+        auto const indices = NumLib::getIndices(id, source_term_dof_table);
+        b.add(indices, _local_rhs);
+    }
+
+private:
+    ParameterLib::Parameter<double> const& _parameter;
+
+    IntegrationMethod const _integration_method;
+    std::vector<
+        IntegrationPointData<NodalRowVectorType>,
+        Eigen::aligned_allocator<IntegrationPointData<NodalRowVectorType>>>
+        _ip_data;
```

Comment:
Changing type to
```suggestion
std::vector<NodalRowVectorType>> _ip_data;
```
would make the `IntegrationPointData` unnecessary...

Reply:
Indeed, good point. Thanks.